crates/amaru-kernel/src/lib.rs

```diff
@@ -750,4 +750,40 @@ mod test {
+            ),
+            (
+                // f93b6be2a0b021c592213184c255d08020a6e9d61701fb42ab2eda65c7025334
+                cbor::decode(hex::decode("845820B546350F727EA033B3A71F7F3D1F490381E6C08AE42654CCD4C5C928D5E136815840DC8D717FC5EE4530E8A688865502C912FDDD630D8A53CED563EC8C70C6AD0B1B9639774BC71784176B87062CC3E6C3AE2A55F5EE4157A5606251C555C8951D025820E621E9CC4FCD59E3321452113DF5EEC8C42214D13CBCD2E886674F0D7B30ECF85825A201581E581C22C0BE3476E5BCC19C53A28B944B1DE988C4A9098428A1F9CBAED86E024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),    hex::decode("323EA3DD5B510B1BD5380B413477179DF9A6DE89027FD817207F32C6").expect("Failed to decode root")
+            ),
+            (
+                // 22e306257262c338dd442936e4dfa84b65b852c60f22bf1224c9a322d13a953e
+                cbor::decode(hex::decode("845820A092633EC3F70B65F7BE2B52134643115E8E848491D1EB074F858FAD9178BE3C584087A5285E3FCBF2B2F118FB6507E83215E8FB4F986071C69232FF9FAD1CAEE7AE30023FC7C42A8AEE87C86DBF33762DF9AAB074014D11323572B9BBF5C744BA0658202B6DD4B0A30410E370E22D6B50A92B30085C3CD920468E660D5977E5008829A844A1024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),
+                hex::decode("232B6238656C07529E08B152F669507E58E2CB7491D0B586D9DBE425").expect("Failed to decode root")
+            ),
+            (
+                // c4b115df82f104e71c173b4f73fd4efc03d87f4e81d6e8b0f2bb67b3ddc90235
+                cbor::decode(hex::decode("845820A6B764333F98CD288DA07626EBCF9D6C6D44C7BE73804030AE704821224D9E515840D0877153B391900CFB82FAEF05194F47D8793610EEDDA6B61C2DA451860E43329B159D4518FE93C9024829F4633C11CED09018258DA8CCC16AD33F9257AC3C0858205755653623B74D85A70B61A3F6B5F8D04701FCF80DBE3C4FADE7B6B554C8754B5825A201581E581C22C0BE3476E5BCE310D2B28B5D21DB7A536889A23C4D3E21F5B35016024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),
+                hex::decode("A5A8B29A838CE9525CE6C329C99DC89A31A7D8AE36A844EEF55D7EB9").expect("Failed to decode root")
+            ),
+            (
+                // 17d3497083063ef23d25280a64657a0efa0b53cc369cc5ef430941f5cc91c94e
+                cbor::decode(hex::decode("8458202B13CBA22AAD6A2B775FDA17F85E7BC74A5CEDB21DDE1D2BE14EC23BE2C9B3C2584006883DEDFFF9C6AF6B4422A81D4D6904C8372E4571D8610007473653722CC07841D5FFA7E427818DD6B9323DE629CF452C5EA08FEA90B2E33E577E26AD8BEF0F58209422C0D6AA61D186C40A0D562911311578463F91C26FA56F3FECD8BBEC0E3A1544A1024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),
+                hex::decode("59F44FD32EE319BCEA9A51E7B84D7C4CB86F7B9B12F337F6CA9E9C85").expect("Failed to decode root")
+            ),
+        ];
+
+        cases.into_iter().for_each(|(witness, expected_root)| {
```

Comment:
Did you mean to create _multiple test cases_ 😄? Note that the [test_case](https://docs.rs/test-case/latest/test_case/) crate is useful for that.